lightning/src/ln/peer_handler.rs

```diff
@@ -44,6 +43,19 @@ use bitcoin::hashes::sha256::Hash as Sha256;
 use bitcoin::hashes::sha256::HashEngine as Sha256Engine;
 use bitcoin::hashes::{HashEngine, Hash};
 
+/// Handler for BOLT1-compliant messages.
+pub trait CustomMessageHandler : wire::CustomMessageReader {
+	/// Called with the message type that was received and the buffer to be read.
+	/// Can return a `MessageHandlingError` if the message could not be handled.
+	fn handle_custom_message(&self, msg: Self::CustomMessage) -> Result<(), LightningError>;
+
+	/// Gets the list of pending messages which were generated by the custom message
+	/// handler, clearing the list in the process. The first tuple element must
+	/// correspond to the intended recipients node ids. If no connection to one of the
+	/// specified node does not exist, the message is simply not sent to it.
+	fn get_and_clear_pending_msg(&self) -> Vec<(Vec<PublicKey>, Self::CustomMessage)>;
```

Comment:
Is there a reason to have a `Vec` of `PublicKey`s here? Isn't the "common" use-case to send a single message to a single peer, not a single message to many peers?

Reply:
Reverted to single `PublicKey`.